skycatalogs/objects/galaxy_object.py

```diff
@@ -161,3 +162,20 @@ def get_instcat_entry(self, band='r', component=None):
         if component not in self.subcomponents:
             return ''
         return form_object_string(self, band, component)
+
+
+class GalaxyConfigFragment(BaseConfigFragment):
+    def __init__(self, prov, cosmology, tophat_bins,
+                 area_partition=None, data_file_type=None):
+        super().__init__(prov, object_type_name='galaxy')
+
+        self._opt_dict = {'area_partition': area_partition,
+                          'data_file_type': data_file_type}
+        self._cosmology = cosmology
+        self._tophat_bins = tophat_bins
+
```

Comment:
Similar to my comment in `DiffskyConfigFragment`, why not just add `Cosmology` and `tophat` entries to `self._opt_dict`:
```
self._opt_dict = {'area_partition': area_partition,
                  'data_file_type': data_file_type,
                  'Cosmology': cosmology,
                  'tophat': {'bins': topat_bins}}
```
Along with the suggestion for `DiffskyConfigFragment`, this would enable all subclasses to use the same `make_fragment` implementation in the base class.

Reply:
I can add the `'Cosmology'` entry to `self._opt_dict` but not `'tophat'` because its value in the template is dict with a couple keys other than `'bins'` which would get lost with the update.